### ISSUE TITLE
Steam Deck controller doesnt work after Lutris flatpak update, possibly runtime related.

### ISSUE THREAD

**Issue by pizzadude** [flathub/net.lutris.Lutris#467]
The Steam Deck controller does not work in Lutris after the latest update to version 0.5.18. I think this is flatpak runtime related because Heroic Games Launcher had the same issue recently, and downgrading the runtime to the 2023 one fixed the issue.

here is the heroic games launcher issue for reference https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/issues/4108

Downgrading the Lutris flatpak with "Warehouse" or with the terminal is the workaround until this gets resolved.

I also posted this issue here, https://github.com/lutris/lutris/issues/5766 , but I was told to post it in this repo also.

**Comment by Eproms** [comment]
Control don't work in game mode for me 😑

**Comment by scottruzal** [comment]
Just started having this issue today as well. Does anyone know what the steps are to downgrade using "Warehouse" or via terminal?

**Comment by pizzadude** [comment]
> Just started having this issue today as well. Does anyone know what the steps are to downgrade using "Warehouse" or via terminal?

Install Warehouse, select Lutris, and then do this:

![image](https://github.com/user-attachments/assets/49fc5c66-5fb9-477f-b047-b3fa2730d52c)

**Comment by mittler-works** [comment]
I have the same problem on the Steam Deck. Downgrading is a temporary workaround. It cannot be the permanent solution.

The problem is that those having this issue may not be the ones who can fix the problem in a professional way. And those who could fix it may not have a steam deck.

So if anyone could provide guidance on how to get startet debugging this flatpak stuff, I'm happy to give it a try.

**Comment by rollercoaster** [comment]
I have the same issue. It was working with existing games, updated lutris via discover, steam deck inbuilt controller no longer working .. I tried a new wine folder with everything default, with both ge-proton latest and wine-ge-8-26

I had been exclusively using wine-ge-8-26 before the updated. After the updated it auto installed ge-proton. When I try this with a game that I know was already working, on launch in game mode, I can hear the audio but the screen still shows the steam loading animation and never shows the game

**Comment by A6GibKm** [comment]
This seems to be an issue related to the runtime, it was updated from GNOME 45 to 47 in a single commit, but the beta branch (at the flathub-beta repo) did have 46 for a while. If anyone wants to contribute to finding the solution I would suggest to first see if the commits using GNOME 46 did have this issue.

You can use
```
flatpak update --commit=<commit_code> <Application ID>
```
to checkout a specific commit. (I don't remember if the commit is the one from github or flatpak. You can use 

```
flatpak remote-info --log <repo_name> <Application ID>
```

to get the list of the latest commits.

I suspect the right place to report this is the freedesktop sdk (https://gitlab.com/freedesktop-sdk) which the GNOME runtime is based on but that depends on whether it is reproducible on 46. Note that 45 and 46 are based on FDO 23.08 while 47 is based on 24.08.

**Comment by mittler-works** [comment]
Sooo I tried it. The latest selectable commit from lutris beta before the update to gnome runtime 47 is:

```
    Commit: 1469ae065b697f6c4d8908f270bc8a2426a942b026b314675ef96f4ec879a0ee
   Subject: I don't understand, don't ask me. (b8ff8b2a)
      Date: 2024-10-13 19:52:06 +0000
```

So I installed it via `sudo flatpak update --commit=1469ae065b697f6c4d8908f270bc8a2426a942b026b314675ef96f4ec879a0ee net.lutris.Lutris/x86_64/beta` and then `sudo flatpak make-current net.lutris.Lutris beta`.

I had masked lutris for the above workaround, so I had to first unmask it: `sudo flatpak --remove mask net.lutris.Lutris`. I noticed, that you can only mask the stable version with `sudo flatpak mask net.lutris.Lutris/x86_64/stable` so it is possible to update beta while preventing updates for stable.

Long story short, with beta commit `1469ae065b697f6c4d8908f270bc8a2426a942b026b314675ef96f4ec879a0ee` (i.e. gnome runtime version 46 it works fine.

**Comment by A6GibKm** [comment]
Thanks, then it is probably an issue in the 24.08 FDO runtime. I will ask around, and maybe do a PR to downgrade to 46 in the meanwhile.

**Comment by mittler-works** [comment]
You're welcome and thank you, too! For the record, I've tried the next commit in history to find more evidence, i.e.

```
    Commit: 3905696721ac81ee885709def81b4b923b6664bd9c5514bc3c21e1e7d2baa8ad
   Subject: Update runtime to GNOME 47 (4c032834)
      Date: 2024-10-13 22:24:40 +0000
```

and it stops working. So it breaks within this diff: https://github.com/flathub/net.lutris.Lutris/compare/b8ff8b2a16561f0b3ce3f9e1f4fce54e16bdbb8c...4c0328342936fb95650e9cec9a37112815559e1a which makes it very likely that it is due to the gnome runtime update.

**Comment by A6GibKm** [comment]
Opened https://github.com/flathub/net.lutris.Lutris/pull/468, please test it once CI finishes and produces a testable build.

**Comment by A6GibKm** [comment]
Would it be possible to open the app via the CLI

```
flatpak run net.lutris.Lutris -d
```

then see if there is anything in the logs that might point to a cause?

**Comment by A6GibKm** [comment]
Has anyone managed to make the controller with certain games / wine version combinations? Or it does not work at all?

**Comment by pizzadude** [comment]
> Has anyone managed to make the controller with certain games / wine version combinations? Or it does not work at all?

From what I've seen, it doesn't work in all.

> Would it be possible to open the app via the CLI
> 
> ```
> flatpak run net.lutris.Lutris -d
> ```
> 
> then see if there is anything in the logs that might point to a cause?

I ran Lutris with that command in game mode, and all I got was a bunch of dxvk debug output for the game I opened (Trine 4).

I ran it in nested desktop (desktop mode within game mode) and this is what it outputted, but I don't think it's anything particularly useful:

```
flatpak run net.lutris.Lutris -d

(process:2): Gdk-WARNING **: 08:12:35.355: Failed to read portal settings: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface ?org.freedesktop.portal.Settings? on object at path /org/freedesktop/portal/desktop

(lutris:2): Gdk-WARNING **: 08:12:35.626: Failed to read portal settings: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface ?org.freedesktop.portal.Settings? on object at path /org/freedesktop/portal/desktop
INFO     2024-12-08 08:12:36,533 [application.do_command_line:480]:Starting Lutris 0.5.18
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
radv: Can't open file: '/tmp/radv_vrs.JtEycGJt'.
radv: Failed to initialize the notifier for RADV_FORCE_VRS_CONFIG_FILE!
INFO     2024-12-08 08:12:36,593 [startup.run_all_checks:120]:"card0" is AMD Custom GPU 0405 (1002:163f 1002:0123 amdgpu) Driver 24.2.7
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
DEBUG    2024-12-08 08:12:36,740 [xrandr.get_resolutions:91]:Retrieving resolution list
DEBUG    2024-12-08 08:12:36,743 [xrandr._get_vidmodes:18]:Retrieving 20 video modes from XrandR
ERROR    2024-12-08 08:12:36,759 [i18n.get_user_locale:11]:Unable to get locale
DEBUG    2024-12-08 08:12:36,803 [path_cache._update_missing_games:114]:Checking for missing games
DEBUG    2024-12-08 08:12:36,841 [api.check_stale_runtime_versions:52]:Modified at Sun Dec  8 08:07:10 2024, will update after Sun Dec  8 14:07:10 2024

(net.lutris.Lutris:2): Gtk-WARNING **: 08:12:36.842: Creating a portal monitor failed: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface ?org.freedesktop.portal.Inhibit? on object at path /org/freedesktop/portal/desktop
DEBUG    2024-12-08 08:12:36,843 [lutriswindow.create_runtime_updater_cb:1358]:Runtime up to date

** (net.lutris.Lutris:2): WARNING **: 08:12:36.856: AT-SPI: Could not obtain desktop path or name

ERROR    2024-12-08 08:12:36,856 [style_manager._call_cb:81]:Error reading color-scheme: g-dbus-error-quark: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.Settings” on object at path /org/freedesktop/portal/desktop (19)
Traceback (most recent call last):
  File "/app/lib/python3.12/site-packages/lutris/style_manager.py", line 74, in _call_cb
    values = obj.call_finish(result)
             ^^^^^^^^^^^^^^^^^^^^^^^
gi.repository.GLib.GError: g-dbus-error-quark: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.Settings” on object at path /org/freedesktop/portal/desktop (19)

** (net.lutris.Lutris:2): WARNING **: 08:12:36.859: atk-bridge: GetRegisteredEvents returned message with unknown signature

** (net.lutris.Lutris:2): WARNING **: 08:12:36.859: atk-bridge: get_device_events_reply: unknown signature
INFO     2024-12-08 08:12:40,279 [application.do_shutdown:1049]:Shutting down Lutris
```

**Comment by MrArdillo** [comment]
My controller works but my game turns black on game mode after initiating a game. The sounds work and the controller seems to work but I can't see anything from the game

**Comment by mittler-works** [comment]
> Opened #468, please test it once CI finishes and produces a testable build.

Works :+1:

**Comment by rollercoaster** [comment]
How long does it usually take for the updated to reach Discover?

**Comment by AlmeidaRW** [comment]
> How long does it usually take for the updated to reach Discover?

Until this change is pushed to the master branch - which it's not currently - you will not see it in the Discover store. You can use the console to install the beta

**Comment by midarikishima** [comment]
> > How long does it usually take for the updated to reach Discover?
> 
> Until this change is pushed to the master branch - which it's not currently - you will not see it in the Discover store. You can use the console to install the beta

Please can i have the steps to carryout this process. Installing the beta.

**Comment by mittler-works** [comment]
> Please can i have the steps to carryout this process. Installing the beta.

See https://discourse.flathub.org/t/how-to-use-flathub-beta/

**Comment by rollercoaster** [comment]
Does the merge into master mean the update is available via Discover?

**Comment by joaociocca** [comment]
I'm using Lutris 0.5.18 (according to Lutris, according to Warehouse I'm on 0.5.17?), trying to play my Amazon copy of Wall World on Steamdeck... but both the Steamdeck's native gamepad and my Nintendo Switch Pro controller won't work on the game.

Anything I can do to troubleshoot this?

**Comment by AlmeidaRW** [comment]
> I'm using Lutris 0.5.18 (according to Lutris, according to Warehouse I'm on 0.5.17?), trying to play my Amazon copy of Wall World on Steamdeck... but both the Steamdeck's native gamepad and my Nintendo Switch Pro controller won't work on the game.
> 
> Anything I can do to troubleshoot this?

In theory the issue with controllers not being recognized in Gaming mode was fixed with the update from 2 weeks ago. If you open your Discover Store, make sure you have it updated (i.e., not showing under Updates).

**Comment by A6GibKm** [comment]
I would highly appreciate if someone could verify whether the issue was fixed. There is a build with the GNOME 48 runtime at https://github.com/flathub/net.lutris.Lutris/pull/484. I think this was fixed a few months ago but I haven't confirmed it myself.

**Comment by AlmeidaRW** [comment]
I can confirm I no longer have that issue

**Comment by A6GibKm** [comment]
Just to confirm, you mean with the build I linked above or with the current stable version at flathub?

**Comment by pizzadude** [comment]
@A6GibKm Hi, that test build breaks system deck input again.

**Comment by AlmeidaRW** [comment]
> Just to confirm, you mean with the build I linked above or with the current stable version at flathub?

My apologies, I misread your question. My answer was indeed referencing the latest stable build, I have not installed any builds/betas.

**Comment by pizzadude** [comment]
I tested the latest Heroic flatpak on the deck, and it has the same issue for older wine/proton versions (8.x and below), however, Proton 9.x and GE-Proton 9.x work fine. (didn't test 10.x yet). Maybe going forward, if we can't find a solution to this issue (why does Bottles work?), since Proton 9.x versions work, maybe we could ignore this issue and just tell people to use Proton 9.x and above. Though we would have to test if that's the case on Lutris also.

**Comment by casasfernando** [comment]
Looking into this with the help from SDL guys and there is progress: https://github.com/libsdl-org/SDL/issues/14410

**Comment by strycore** [comment]
Yes, this can be closed.
